---
title: "rnaCountSeb"
author: "Marta Trusohamn"
output: word_document

---
```{r install packages, eval=FALSE, include=FALSE}
#install.packages("plyr")
#install.packages("ggplot2")
#install.packages("ggsignif")
#install.packages("gridExtra")
#install.packages("reshape")
#for griding plots:
#if(!require(devtools)) install.packages("devtools")
#devtools::install_github("kassambara/ggpubr")
#install.packages("ggpubr")
#install.packages("gridGraphics")
```

```{r setup, eval=FALSE, include=FALSE}
rm(list=ls()) 
# wd = "C:/Users/Marta/Desktop/sebastian/expression"
#wd = "C:/Users/mlecz_000/Desktop/sebastian/expression"
#wd = ("C:/Users/Marta/Desktop/Sebastian/expression")
#wd <- "C:/Users/martal1/ownCloud/Desktop/sebastian/expression"
wd = "C:/Users/Marta/Documents/sebastian"

```

```{r dataLoad1, echo=FALSE, include=FALSE, eval = TRUE}
setwd(wd)

###load data
load("merged_ForSeb.Rda")
myFullData <- merge(x =merged$annot , y = t(merged$exp), by.y = "row.names" , by.x="cell_id" , all = TRUE)
rm(merged)
#table(myFullData$level1class)

###factor and change order of cell types
myFullData$level1class <- factor(myFullData$level1class, 
                                 c("Vascular and Leptomeningeal Cells",
                                   "Vascular Smooth Muscle Cell",
                                   "Vascular Endothelial",
                                   "Pericytes",
                                   "Microglia",
                                   "Astrocytes",
                                   "Interneurons",
                                   "Pyramidal Neurons",
                                   "Oligodendrocytes",
                                   "Oligodendrocyte Precursor"))


###rename some of cell types
library(plyr)
myFullData$level1class <- revalue(myFullData$level1class, 
                                  c("Vascular and Leptomeningeal Cells"="VLMC", 
                                    "Vascular Smooth Muscle Cell"="vSMC",
                                    "Vascular Endothelial"="Endothelium",
                                    "Oligodendrocyte Precursor"="OPC"))
```

```{r dataLoad1CB, echo=FALSE, include=FALSE, eval = TRUE}
wdB <- paste0(wd,"/Betsholtz")
setwd(wdB)

###join cells from two datasets, remove gens which are not covered in both
dfB1 <- read.table("GSE98816_Brain_samples_normalized_counts_matrix.txt", header=TRUE, row.names=1, strip.white=TRUE)
dfB2 <- read.table("GSE99058_Brain_AC_250samples_normalized_counts_matrix.txt", header=TRUE, row.names=1, strip.white=TRUE)
dfB <- merge(dfB1, dfB2, by="row.names", all = F)
row.names(dfB) <-dfB$Row.names
dfB$Row.names <-NULL

###read file with cell clusters for dfB1
cellClusters <- read.csv("Betsholtz cell cluster identity brain_3186cell_15clusters.csv")
colnames(cellClusters)[colnames(cellClusters) == 'cell'] <- 'cell_id'
###create df with clusters for dfB2 by regex
cellClusters2 <- data.frame(cell_id = colnames(dfB2), 
                            cluster = regmatches(colnames(dfB2), regexpr("V.*(?=_)", colnames(dfB2), perl=TRUE)))
###add 250 cells with clusters
cellClusters <- rbind(cellClusters, cellClusters2)
###get rid of whitespaces and bring everyting to uppercase
cellClusters$cluster <- toupper(trimws(cellClusters$cluster, which="both")) 

###add column with level1class to the clusters
clustersToClasses <- read.csv("C:/Users/Marta/rnaSeq/CellClustersToClasses.csv")
clustersToClasses$cluster <- toupper(trimws(clustersToClasses$cluster, which="both"))
clustersToClasses$level1class <- trimws(clustersToClasses$level1class, which="both")
cellClusters <- merge(x =cellClusters, y = clustersToClasses, by.y = "cluster" , by.x="cluster" , all = TRUE)

myFullData2 <- merge(x =cellClusters, y = t(dfB), by.y = "row.names" , by.x="cell_id" , all = TRUE)
###factor and change order of cell types
myFullData2$level1class <- factor(myFullData2$level1class, 
                                 c("VLMC",
                                   "vSMC",
                                   "Endothelial",
                                   "Pericytes",
                                   "Microglia",
                                   "Astrocytes",
                                   "Oligodendrocytes"))


```

```{r dataLoad2, echo=FALSE, include=FALSE, eval = TRUE} 
setwd(wd)

###load data
df <- read.csv("GSE18920_for_R_wGeneNames (no log2).csv")

###show gene duplicates
# n_occur <-  data.frame(table(df$NAME))
# duplic_genes<-n_occur[n_occur$Freq > 1,]
# duplic_values <- df[duplicated(df$NAME),]

###removing the second occurance
df<-df[!duplicated(df$NAME),]

arrangeDf <- function(df){
  ###transposing df
  row.names(df)<-df$NAME
  df$NAME=NULL
  df<-t(df)
  
  ###getting informations from column names
  Tissue <- c()
  Phen <- c()
  Id_pat <- c()
  for (rname in rownames(df)){
    fullname <- strsplit(rname, "_")
    Tissue <- c(Tissue, fullname[[1]][1])
    Phen <- c(Phen,fullname[[1]][2])
    Id_pat <- c(Id_pat, fullname[[1]][3])
  }
  ###adding description rows to df
  desc <- data.frame(row.names=rownames(df), Tissue=Tissue, Phen=Phen, Id_pat=Id_pat)
  df <- merge(y = df, x = desc, by.x = "row.names" , by.y="row.names" , all = TRUE)
  
  row.names(df)=df$Row.names
  df$Row.names=NULL
  
  ###removing spaces
  names(df) <- gsub(" ", "", names(df))
  df$Phen <- factor(df$Phen, c("CTRL", "ALS"))
  return(df)
}

df <- arrangeDf(df)

###keeping only AH
df <-subset(df, Tissue=="AH")
```

```{r dataLoad3, echo=FALSE, include=FALSE, eval = TRUE} 
setwd(wd)
df2 <- read.csv("GSE18597_data_firstName.csv")
###name of 1 column -> NAME
names(df2)[1]<-paste("NAME")

###check for duplicates
#n_occur <-  data.frame(table(df2$NAME))
#duplic_genes<-n_occur[n_occur$Freq > 1,]

###remove the second occurance
df2<-df2[!duplicated(df2$NAME),]


arrangeDf2 <- function(df2){
  ###transpose df
  row.names(df2)<-df2$NAME
  df2$NAME=NULL
  df2$X=NULL
  df2<-t(df2)
  ###adding description rows
  Id <- c()
  Genotype <- c()
  Day <- c()
  MP <- c()
  for ( rname in rownames(df2)){
    fullname <- strsplit(rname, "_")
    Id <- c(Id, fullname[[1]][1])
    part<-strsplit(fullname[[1]][3], "\\.")
    Genotype <- c(Genotype,substr(fullname[[1]][3],1,3))
    Day <- c(Day, substr(part[[1]][1],4, (nchar(part[[1]][1])-1)) )
    MP <- c(MP, part[[1]][2] )
  }
  desc2 <- data.frame(row.names = rownames(df2), Id=Id, Genotype=Genotype, Day=Day, MP=MP)
  df2 <- merge(y = df2, x = desc2, by.x = "row.names" , by.y="row.names" , all = TRUE)
  row.names(df2)=df2$Row.names
  df2$Row.names=NULL
  df2$Day <- as.numeric(as.character(df2$Day))
  return(df2)
}

df2 <- arrangeDf2(df2)

```



```{r loadPlots, echo=FALSE, dpi=300, fig.width=5, fig.height=8, dev.args=list(pointsize=16)}

library(ggplot2)

#################
#Spec plot
#################
    
  color10 <- c("#EC1C24","#FAAF40","#FFD700", "#A0522D", "#2E8B57","#6DC7BE", "#7F3F97","#524FA0", "#BBBDBF", "#808184")
plot1 <- function(gen, dataset = myFullData, color=color10){

  avals <- c(rep.int(0.4,10))
  avalsHex <- paste0(color, toupper(as.hexmode(round(avals*255))))
  
  ###create a temp dataframe with only gen column
  temp <- dataset[c(gen, "level1class")]
  names(temp)[1] <-"gen"
  
  ggplot(temp, aes(x=level1class , y=gen)) +
    geom_boxplot(aes(fill=level1class, color=level1class), outlier.colour = NA, alpha = 0.65)+
    ###cover colorfull mean crossbar with white strap
    stat_summary(geom = "crossbar", width=1, fatten=0, size=0.75, color="white",
                fun.data = function(x){ return(c(y=median(x), ymin=median(x), ymax=median(x))) })+
    scale_fill_manual(values=color)+
    scale_alpha_manual(values = avals) +
    scale_colour_manual(values = avalsHex)+
    geom_jitter(aes(color=level1class ), size=1, alpha=0.3, position = position_jitter(width = .05))+
    theme(panel.background = element_rect(fill='white', colour='white'),
          legend.position="none",legend.title=element_blank(),
          axis.line.y =element_line(colour="grey85"), 
          axis.title.y=element_text(colour="grey45"), 
          axis.text.y = element_text(colour="grey45"), 
          axis.ticks.y = element_line(colour="grey45"),
          axis.text.x=element_text(colour="grey45",angle = 90, vjust=0.25, hjust=1, size=8),
          axis.ticks.x = element_blank(), 
          axis.line.x = element_line(colour="grey85"), 
          axis.title.x=element_blank())+
    xlab("")+
    ylab("RNA count per cell")+
    ggtitle(gen)+
    labs(subtitle=("Cell type specific expression"))
}

##############
#time box plot
##############

###red and blue
color <- (c("#01abff", "#d82101"))


data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  return(data_sum)
}

plot2<-function(gen){
  ###transparent colors
  avals <- c(rep.int(0.3,2))
  avalsHex <- paste0(color, toupper(as.hexmode(round(avals*255))))
  
  ###create a temp dataframe with only gen column
  temp <- df2[c(gen, "Genotype", "Day")]
  names(temp)[1] <-"gen"
  ds <- data_summary(temp, varname="gen", groupnames=c("Genotype", "Day"))
  
  ggplot(temp, aes_string(x=factor(temp$Day, c("28","42","56","70","98","112","126")), y="gen") ) + 
    #geom_line(size=1.5) +
    geom_boxplot(aes(fill=Genotype, color=Genotype), varwidth=TRUE, position = position_dodge(width=0),
                 alpha=0.5)+
    scale_fill_manual(values=color)+
    scale_alpha_manual(values = avals) +
    scale_colour_manual(values = avalsHex)+
    stat_summary(aes(group=Genotype),geom = "crossbar", width=1, fatten=0, size=0.75, color="white", 
                 fun.data = function(x){ return(c(y=median(x), ymin=median(x), ymax=median(x))) })+
    geom_jitter(aes(color=Genotype ), size=1, alpha=0.9, position = position_jitter(width = 0))+
    ylab("RNA expression levels")+
    xlab("Day")+
    ggtitle("")+
    labs(subtitle=("Mouse SOD1 (G93A)"))+
    theme(panel.background = element_rect(fill='white', colour='white'),
          legend.position="right",legend.title=element_blank(),
          axis.line.y = element_line(colour="grey85"), axis.title.y=element_text(colour="grey45"), 
          axis.text.y = element_text(colour="grey45"), axis.ticks.y = element_line(colour="grey45"),
          axis.text.x=element_text(colour="grey45"), axis.ticks.x = element_line(colour="grey45"), 
          axis.line.x = element_line(colour="grey85"), axis.title.x=element_text(colour="grey45"),
          legend.text = element_text(colour="grey45")) +
    stat_summary(fun.y=mean, geom="smooth", aes(group=Genotype, color=Genotype))
  
}

###############
#boxplot human
###############
plot3 <- function(gen){
  genUP <-toupper(gen)
  #transparent colors
  avals <- c(rep(0.3,2))
  avalsHex <- paste0(color, toupper(as.hexmode(round(avals*255))))
  
  #create a temp dataframe with only "gen"" column
  temp <- df[c(genUP, "Tissue", "Phen")]
  names(temp)[1] <-"gen"
  
  ggplot(temp, aes(x=Phen, y=gen)) +
    geom_boxplot(aes(fill=Phen, color=Phen), outlier.colour = NA, alpha=0.5)+
    scale_fill_manual(values=color)+
    scale_alpha_manual(values = avals) +
    scale_colour_manual(values = avalsHex)+
    ###white mean stripe in boxplot instead of colorful one
    stat_summary(geom = "crossbar", width=1, fatten=0, size=0.75, color="white", 
                 fun.data = function(x){ return(c(y=median(x), ymin=median(x), ymax=median(x))) })+
    geom_jitter(aes(group=Phen, color=Phen), size=1, alpha=0.9, position = position_jitter(width = .05))+
    xlab(" ")+
    ylab("RNA expression levels")+
    ggtitle("")+
    labs(subtitle=("Human ALS"))+
    theme(panel.background = element_rect(fill='white', colour='white'),
          legend.position="none",legend.title=element_blank(),
          axis.line.y = element_line(colour="grey85"), 
          axis.title.y=element_text(colour="grey45"), 
          axis.text.y = element_text(colour="grey45"), 
          axis.ticks.y = element_line(colour="grey45"),
          axis.text.x=element_text(colour = "grey45"), 
          axis.ticks.x = element_blank(), 
          axis.line.x = element_line(colour="grey85"), 
          axis.title.x=element_text(colour = 'white'),
          strip.text = element_text(colour = 'white', face="bold"), 
          strip.background =element_rect(fill="darkgray"))
}

##########
#plot3 for website (if human name is different than mouse)
##########
plot3_withName <- function(gen){
  genUP <-gen
  #transparent colors
  avals <- c(rep(0.3,2))
  avalsHex <- paste0(color, toupper(as.hexmode(round(avals*255))))
  
  #create a temp dataframe with only "gen"" column
  temp <- df[c(genUP, "Tissue", "Phen")]
  names(temp)[1] <-"gen"
  
  ggplot(temp, aes(x=Phen, y=gen)) +
    geom_boxplot(aes(fill=Phen, color=Phen), outlier.colour = NA, alpha=0.5)+
    scale_fill_manual(values=color)+
    scale_alpha_manual(values = avals) +
    scale_colour_manual(values = avalsHex)+
    ###white mean stripe in boxplot instead of colorful one
    stat_summary(geom = "crossbar", width=1, fatten=0, size=0.75, color="white", 
                 fun.data = function(x){ return(c(y=median(x), ymin=median(x), ymax=median(x))) })+
    geom_jitter(aes(group=Phen, color=Phen), size=1, alpha=0.9, position = position_jitter(width = .05))+
    xlab(" ")+
    ylab("RNA expression levels")+
    ggtitle("")+
    labs(title = gen, subtitle=("Human sALS"))+
    theme(panel.background = element_rect(fill='white', colour='white'),
          legend.position="none",legend.title=element_blank(),
          axis.line.y = element_line(colour="grey85"), 
          axis.title.y=element_text(colour="grey45"), 
          axis.text.y = element_text(colour="grey45"), 
          axis.ticks.y = element_line(colour="grey45"),
          axis.text.x=element_text(colour = "grey45"), 
          axis.ticks.x = element_blank(), 
          axis.line.x = element_line(colour="grey85"), 
          axis.title.x=element_text(colour = 'white'),
          strip.text = element_text(colour = 'white', face="bold"), 
          strip.background =element_rect(fill="darkgray"))
}

###generate an empty plot
empty <- ggplot() + geom_point(aes(1, 1), colour = "white") + 
  theme(plot.background = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_blank(), 
        panel.background = element_blank(), 
        axis.title.x = element_blank(),
        axis.title.y = element_blank(), 
        axis.text.x = element_blank(), 
        axis.text.y = element_blank(),
        axis.ticks = element_blank())


```

```{r printForSingleGene, echo=FALSE}
gen="Rgs9bp"
#homolog <- as.character(renames[match(gen,renames$mouse), "human"])
print(plot1(gen, myFullData2, color10))

print(plot2(gen))
print(plot3(gen))
print(plot3_withName(toupper(gen)))
```

```{r prepareDataForDrawing, echo=FALSE, eval=FALSE, include=FALSE}
###Run before generating the figures for website

###rename to Ctrl and ALS
df2$Genotype <- revalue(df2$Genotype, c("Ctr" = "Ctrl", "SOD" = "ALS"))
df$Phen <- revalue(df$Phen, c("CTRL" = "Ctrl", "ALS" = "ALS"))


###load rename list (mouse to humans names)
setwd(wd) 
renames<- read.csv("mouse_human_one_to_one.csv")




```

```{r drawPlotsForWebsite}

generateGraphFile <- function(gen){
  #print(gen)
    filename = gen
    #spec
    if (gen %in% colnames(myFullData)) p1 <- plot1(gen) else p1 <- empty
    
    #dynamics
    if (gen %in% colnames(df2)) p2 <- plot2(gen) else p2 <- empty
    
    #human
    if (toupper(gen) %in% colnames(df)) {
      p3 <- plot3(gen)
    }else if (gen %in% renames$mouse){
      homolog <- as.character(renames[match(gen,renames$mouse), "human"])
      
      if(homolog %in% colnames(df)) p3 <- plot3_withName(homolog)
      else p3 <- empty
      
      filename2 = tolower(homolog)
      png(file=paste0(filename2, ".png"), res=325, width=2500, height = 1750) 
      grid.arrange(p1, p2, p3, nrow = 1, widths = c(5,7,3))
      dev.off()
      
    }else p3 <- empty
    
    filename = tolower(filename)
    png(file=paste0(filename, ".png"), res=325, width=2500, height = 1750) 
    grid.arrange(p1, p2, p3, layout_matrix = lay, widths= c(4,4,2,12,2,6), heights= c(8,1))
    dev.off()
}

setwd(wd)

lay <- rbind(c(1,1,4,2,4,3),
             c(1,1,4,5,4,5))
library(gridExtra)

#genes <- c("Prn")
genes <- colnames(myFullData)[(100:120)] 
#genes <- colnames(myFullData)[-(1:20980)] 

lapply(genes, generateGraphFile)

```


```{r figureSpec, echo=FALSE, dpi=300, fig.width=5, fig.height=8, dev.args=list(pointsize=16)}
library(reshape)
library(ggplot2)
## Gives count, mean, standard deviation, standard error of the mean, and confidence interval (default 95%).
##   data: a data frame.
##   measurevar: the name of a column that contains the variable to be summariezed
##   groupvars: a vector containing names of columns that contain grouping variables
##   na.rm: a boolean that indicates whether to ignore NA's
##   conf.interval: the percent range of the confidence interval (default is 95%)
#http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum <- ddply(data, groupnames, .fun=summary_func, varname)
  return(data_sum)
}

summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE, conf.interval=.95, .drop=TRUE) {
  library(plyr)
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar)
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

theme1 <- theme(panel.background = element_rect(fill='white', colour='white'),
                legend.position="none",
                legend.title=element_blank(),
                axis.line.y = element_blank(), 
                axis.title.y=element_blank(), 
                axis.text.y = element_text(colour="grey45"), 
                axis.ticks.y = element_line(colour="grey45"),
                axis.text.x=element_text(colour="grey45",angle = 90, hjust = 1, size=8),
                axis.ticks.x = element_blank(), 
                axis.line.x = element_blank(), 
                axis.title.x=element_blank())

theme2 <- theme(panel.background = element_rect(fill='white', colour='white'),
                legend.position="none",
                legend.title=element_blank(),
                axis.line.y = element_blank(), 
                axis.title.y=element_blank(), 
                axis.text.y =  element_blank(), 
                axis.ticks.y =  element_blank(),
                axis.text.x=element_text(colour="grey45",angle = 90, hjust = 1, size=8),
                axis.ticks.x = element_blank(), 
                axis.line.x = element_blank(), 
                axis.title.x=element_blank(),
                strip.text.y = element_text(size = 5))

colorb10 <- c("#EC1C24","#FAAF40","#FFD700", "#A0522D", "#2E8B57","#6DC7BE", "#7F3F97","#524FA0", "#BBBDBF", "#808184")


fig1plot  <- function(genes, dataset = myFullData, color=colorb10){
  library(reshape)
  
  avals <- c(rep.int(0.4,10))
  avalsHex <- paste0(color, toupper(as.hexmode(round(avals*255))))
  
  name <- genes[1] #first value is a name of a group
  genes <- genes[(-1)]
  
  colnames <- c("level1class" ,  genes)
  sampleData <- melt(dataset[, colnames])
  
  dsf <- data_summary(sampleData, varname="value", groupnames= c("variable", "level1class" )) #SD
  dsfse <- summarySE(sampleData, measurevar="value", groupvars= c("variable", "level1class" )) #SE
  
  
  library(ggplot2)
  ggplot(dsfse, aes(x=level1class , y= value, fill=level1class)) +
    geom_errorbar(aes(ymin=value, ymax=value+se), width=.2,
                  position=position_dodge(.9))+
    geom_bar(stat="identity", 
             position=position_dodge(), colour="white")+
    scale_fill_manual(values=color)+
    theme2+
    facet_grid(variable ~., scales="free")+
    xlab("")+
    ylab("counts")+
    ggtitle(name)+
    labs(subtitle=("Mean with SEM"))
}



sets <- 
#list(c("VSMC UP 2018.08.21","Ascl2","Alx1","Mylk", "Flna", "Palld", "Fosl2", "Prss23", "Crlf1"))
#list(c("VSMC DOWN 2018.08.21","Des","Amigo2","Nexn","Tmem38b","Sh3bgr", "Zfp385c","Nrarp"))
#list(c("EC DOWN","Pvr","Pecam1","Rbl1","8430408G22Rik","Rgs9bp","Tmem204","Ccm2l","Tdrp","Efna3","Sgpp2"))
#list(c("EC UP", "Ccdc141","Pgm5","Icam1","Hhex","Timeless","Capg","Cd34","Tnfrsf10b","Il2rg","Klhl6"))
list(c("PC Down", "Saa1","Ebf1","Abcc9","Heyl","Aif1l","Nrarp"))
#list(c("PC UP", "Sik1","S1pr3","Atp13a5","Fam161a","Rarres2","Gpx8","Lamc1","Arhgdib","Ggta1","Plxdc2" ))

setwd(wd)
library(gridExtra)
pdf(file="fig1edataLin.pdf",  width=6, height=16) 
s1=fig1plot(sets[[1]], myFullData)
grid.arrange(s1, empty, empty, ncol=1)
dev.off()

# #pdf(file="fig1.pdf",  width=4, height=28) 
# s1=fig1plot(sets[[1]], myFullData2)
# s2=fig1plot(sets[[2]], myFullData2)
# 
# grid.arrange(s1, s2, ncol=1)
# #dev.off()

# 
# setwd(wd)
# library(gridExtra)
# pdf(file="fig1.pdf",  width=4, height=28) 
# s1=fig1plot(sets[[1]])
# s2=fig1plot(sets[[2]])
# #s3=fig1plot(sets[[3]])
# #grid.arrange(s1, s2, s3, ncol=1)
# grid.arrange(s1, s2, ncol=1)
# dev.off()

```


```{r figureDynamics, echo=FALSE, dpi=300, fig.width=5, fig.height=8, dev.args=list(pointsize=16)}

sets <- list(c("Early", "Col1a1", "Col1a2","Col15a1",  "Bgn", 'Col3a1', "Igf2","Col6a3"),
             c("Sustained", "Lum","Col5a2","Apod", "Lama2","Pcolce", "Nid1" ),
             c("Late",  "Nupr1","Ccl11","Igfbp2",  "Serping1", "Spp1", "Dcn", "Itih5"  )
) 

fig2plot <- function(aSet){
  
  subdata <-df2[, c("Genotype", "Day", aSet[-1])]
  subdata$Day <- as.factor(subdata$Day)
  
  
  sumTable = data.frame(Gene=character(), Day=numeric(), meanCNTRL=numeric(), meanSOD=numeric(), diff=numeric(), p.value=numeric())
  
  for(gene in colnames(subdata)[-c(1,2)]){
    for(day in subdata$Day){
      cntrl <- subdata[subdata$Day==day & subdata$Genotype == 'Ctr', gene]
      sod <- subdata[subdata$Day==day & subdata$Genotype == 'SOD', gene]
      t<- unlist(t.test(cntrl,sod))
      p.value <- t[3]
      cntrl.value <- t[6]
      sod.value <- t[7]
      diff <- as.numeric(sod.value) / as.numeric(cntrl.value)
      entry <- data.frame (Gene=gene, Day=day, meanCNTRL=cntrl.value, meanSOD=sod.value, diff=diff, p.value=p.value)
      sumTable = rbind(sumTable, entry)
    }
  }
  sumTable$Day <- as.numeric(as.character(sumTable$Day))
  sumTable$p.value <- as.numeric(as.character(sumTable$p.value))
  sumTable$Signif <- 0
  sumTable$Signif[sumTable$p.value < 0.05] <- 1
  sumTable$Signif <- as.factor(sumTable$Signif)
  
  library(ggplot2)
  ggplot(sumTable, aes(x=Day, y=diff, color=Gene))+
    geom_line()+
    geom_point(aes(size=Signif))+
    geom_hline(yintercept=1, linetype="dashed", color = "grey45", size=0.4)+
    scale_size_manual(values=c(0,2))+
    guides(size=FALSE)+
    expand_limits(y=0)+
    scale_x_continuous(breaks = c(28, 42, 56,  70,  98, 112 ,126))+
    ggtitle(aSet[1])+
    scale_color_brewer(palette="Spectral")+
    theme(panel.background = element_rect(fill='white', colour='white'),
          legend.position="right",legend.title=element_blank(),
          axis.line.y = element_line(colour="grey45"), axis.title.y=element_blank(), axis.text.y = element_text(colour="grey15"), axis.ticks.y = element_line(colour="grey15"),
          axis.text.x=element_text(colour="grey15"), axis.ticks.x = element_line(colour="grey15"), axis.line.x = element_line(colour="grey45"), axis.title.x=element_text(colour="grey45"),
          legend.text = element_text(colour="grey45"))
}

setwd(wd)
library(gridExtra)
pdf(file="fig2.pdf",  width=4, height=8) 
s1=fig2plot(sets[[1]])
s2=fig2plot(sets[[2]])
s3=fig2plot(sets[[3]])
grid.arrange(s1, s2, s3, ncol=1)
dev.off()

```
